### PR TITLE
13 one minus kaplan meier

### LINKED ISSUE #13: one minus kaplan meier
- [x] change name to one_minus_kaplan_meier

- [x]  use snake_case for variable

 

- [x]  add roxygen2 documentation

- [x]  used checkmate package for assertions of arguments (see e.g. https://github.com/openpharma/mmrm/blob/11e3b2d0f28cf98a526843f240f8ae8a6c79a92d/R/fit.R#L30 as example)

 

- [x]  added to _pkgdown.yml if exported

- [x]   added examples if exported

### PR DESCRIPTION
**Pull request by kuenzelt**
closes #13

### CODE REVIEW

File: R/prop_trans_inc_dens_ce.R [hunk 1]
```diff
@@ -1,8 +1,19 @@
 #' Probability Transform Incidence Density Accounting for Competing Events
 #'
+<<<<<<< HEAD
```
Comment (danielinteractive) [inline R/prop_trans_inc_dens_ce.R:3]: there is a merge conflict here, please can you edit this to fix it?

File: R/one_minus_kaplan_meier.R [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#' One Minus Kaplan-Meier
+#'
+#' This funciton calculates the one minus Kaplan-Meier estimator of adverse events (while censoring all competing events)
```
Comment (danielinteractive) [inline R/one_minus_kaplan_meier.R:3]:
```suggestion
#' This function calculates the one minus Kaplan-Meier estimator of adverse events (while censoring all competing events)
```

File: R/one_minus_kaplan_meier.R [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+#' observed in `[0, tau]`.
+#' Please also refer to formula (4) in
+#'
+#' Stegherr R, Beyersmann J, Jehl V, Rufibach K, Leverkus F, Schmoor C, Friede T.
```
Comment (danielinteractive) [inline R/one_minus_kaplan_meier.R:7]: please use the references syntax as e.g. in
https://github.com/openpharma/savvyr/blob/main/R/prop_trans_inc_dens_ce.R#L26

File: R/one_minus_kaplan_meier.R [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  if (nrow(data %>% filter(type_of_event == 1)) == 0) {
+    ae_prob <- 0
+    ae_prob_var <- 0
+  } else {
+    help <- data.frame(id = data$id)
+    help$from <- 0
+    help$to <- ifelse(data$type_of_event != 1, "cens", data$type_of_event)
+    help$time <- ifelse(data$time_to_event == 0, 0.001, data$time_to_event)
+
+    tra <- matrix(FALSE, 2, 2)
```
Comment (danielinteractive) [inline R/one_minus_kaplan_meier.R:50]: consider better variable name so it is clearer what this is

File: R/prop_trans_inc_dens_ce.R [hunk 5]
```diff
@@ -23,9 +34,12 @@
 #'
 #' @export
 #'
+<<<<<<< HEAD
```
Comment (danielinteractive) [inline R/prop_trans_inc_dens_ce.R:37]: also here

File: tests/testthat/test-one_minus_kaplan_meier.R [hunk 6]
```diff
@@ -0,0 +1,13 @@
+test_that("one_minus_kaplan_meier works as expected", {
```
Comment (danielinteractive) [inline tests/testthat/test-one_minus_kaplan_meier.R:1]: can you add a second test that looks into the edge case where you 0 rows and just return 0?

**Review by danielinteractive — COMMENTED** [review]
Thanks a lot @kuenzelt ! Please see below
